syntax = "proto3";

package todo;

// Definição das mensagens
message TodoItem {
  int32 id = 1;
  string title = 2;
  string description = 3;
  bool completed = 4;
}

message Empty {}

message TodoId {
  int32 id = 1;
}

message TodoList {
  repeated TodoItem items = 1;
}

// Serviço gRPC
service TodoService {
  rpc AddTodo (TodoItem) returns (TodoItem);
  rpc GetTodos (Empty) returns (TodoList);
  rpc CompleteTodo (TodoId) returns (TodoItem);
  rpc DeleteTodo (TodoId) returns (TodoItem);
}
